compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -2713,6 +2714,42 @@ private void checkConstantAssignment(BLangExpression varRef, AnalyzerData data)
         }
     }
 
+    private boolean isFiniteWithNilMember(BType bType) {
+        if (!(bType.getKind() == TypeKind.FINITE)) {
+            return false;
+        }
+        BFiniteType finiteType = (BFiniteType) bType;
+        Set<BLangExpression> finiteMembers = finiteType.getValueSpace();
+        for (BLangExpression finiteMemberType : finiteMembers) {
+            if (finiteMemberType.getBType() != null && finiteMemberType.getBType().tag != TypeTags.NIL) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private boolean isUnionWithSingleNilTypeMember(BType bType) {
```

Comment:
We need to add a few negative test including return types such as 

- `()|int|()`
- `1|null`
- `null|1|null`

Reply:
[5bfff30](https://github.com/ballerina-platform/ballerina-lang/pull/39819/commits/5bfff3057a80394569bda7a41c4fbb5414d07e9b)